### DISCUSSION TITLE
Xcode couldn't find any iOS App Development provisioning profiles matching

### DISCUSSION THREAD

**Discussion by unbam** [codemagic-ci-cd/codemagic-docs#1967] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1.  Turned on Automatically manage signing in XCode settings
2. Select release build for iOS in Codemagic build settings
3. iOS code signing turned on in CodeMagic distribution settings
4. Select automatic as code signing method
5. Select Ad hoc for Provisioning profile type
6. Select the Bundle identifier registered in Certificates, Identifiers & Profiles of AppleDeveloper

### Expected results

Ad hoc Provisioning Profile must be found at the time of Building iOS.

### Actual results

Did not find matching provisioning profiles for code signing
But Set up code signing identities is successful.

### Set up code signing identities
![スクリーンショット 2023-08-14 11 39 49](https://github.com/codemagic-ci-cd/codemagic-docs/assets/16283613/46f33271-1fc2-4e8b-865a-e511d39e5a56)

### Buid iOS
![スクリーンショット 2023-08-14 11 36 23](https://github.com/codemagic-ci-cd/codemagic-docs/assets/16283613/e532db42-afaf-4d81-9a9b-c2693ac56423)

**Comment by unbam** [comment]
Additional Information.
* I have created two workflows, production and staging, and produciton is successful.
* production is succeeding with Provisioning profile type set to App store.
* staging is having an Ad hoc "Xcode couldn't find any iOS App Development provisioning profiles matching" error.

```bash
Error (Xcode): No profiles for 'xx.xxxxx.xxxxx.xxxxx.stg' were found: Xcode couldn't find any iOS App Development provisioning profiles matching 'xx.xxxxx.xxxxx.xxxxx.stg'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild.
/Users/builder/clone/app/ios/Runner.xcodeproj

```

**Comment by unbam** [comment]
Self resolved.

**Reply by noorbhatia** [reply]
hey @unbam how did you fix it? I am facing the same issue

**Reply by unbam** [reply]
@noorbhatia 
Code in Codemagic's PreBuild

```bash
find **/*.xcodeproj -type f | xargs sed -i "" -E 's/PRODUCT_BUNDLE_IDENTIFIER = ". +";/PRODUCT_BUNDLE_IDENTIFIER = jp.co.example.app.stg;/g'
```

`jp.co.example.app.stg` specifies your Identifier.
